src/me/mrCookieSlime/Slimefun/AncientAltar/RitualAnimation.java

```diff
@@ -38,11 +45,17 @@ public RitualAnimation(List<Block> altars, Block altar, Location drop, ItemStack
 
 		this.running = true;
 		this.stage = 0;
+		for(Block ped:this.pedestals) {
+			Item itm = AncientAltarListener.findItem(ped);
+			this.itemLock.put(itm, itm.getLocation().clone());
+		}
 	}
 
 	@Override
 	public void run() {
 		idle();
+		if(!checkLockedItems())
+			abort();
```

Comment:
You should probably also add a return; over here, since we don't want to continue anyway.